### PR TITLE
Rating

### PR DESCRIPTION
**Pull request by therese28**
Not completed, saving for the day

### CODE REVIEW

**Comment by RipNasty** [comment]
We made the "Rating" component in our form a Slider.  We also updated our Comment Tiles to display the rating, and applied some appropriate styling to get the slider bar and button to appear.

**Comment by RipNasty** [comment]
https://www.loom.com/share/71b58f06e46641cda0d3cecd138f843b

**Comment by Chepenik** [comment]
Worked on this with David. Very proud of the slider we created with an emoji. Eagerly awaiting comments why we can't use an emoji in the slide bar, but still have some slight hope we can :D. Can we?

**Comment by AngeC22** [comment]
looks good

File: client/src/components/SocialMediaShow.js [hunk 1]
```diff
@@ -10,32 +10,35 @@ const SocialMediaShow = (props) => {
...
+        const errorMessage = `${response.status} (${response.statusText})`
+        const error = new Error(errorMessage)
+        throw error
       }
-      const body = await response.json();
-      setSocialMediaShow(body.website);
-      setComments(body.website.comments);
+      const body = await response.json()
+      setSocialMediaShow(body.website)
+      setComments(body.website.comments)
```
Comment (dovakeen118) [inline client/src/components/SocialMediaShow.js:28]: consider moving `comments` array state to the state for the `website`:
```js
setSocialMediaShow(body.website) // sets state with website AND the `comments`
setComments(body.website.comments) // sets state for just the comments
```

File: client/src/components/SocialMediaShow.js [hunk 2]
```diff
@@ -10,32 +10,35 @@ const SocialMediaShow = (props) => {
     description: "",
   });
 
-  const [comments, setComments] = useState([]);
+  const [comments, setComments] = useState([])
+  const [newComment, setNewComment] = useState({ rating: "", comment: "" })
```
Comment (dovakeen118) [inline client/src/components/SocialMediaShow.js:14]: this state doesn't look like it's being used? the CommentForm defines a very similar piece of state. With forms, we ideally want the state for the `<form>` to be defined in the same component

File: server/src/db/migrations/20230505171025_addRating.cjs [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/**
+ * @typedef {import("knex")} Knex
+ */
+
+/**
+ * @param {Knex} knex
+ */
+exports.up = async (knex) => {
+    return knex.schema.alterTable("comments", (table) => {
+        table.integer("rating").notNullable().defaultTo(0)
```
Comment (dovakeen118) [inline server/src/db/migrations/20230505171025_addRating.cjs:10]: if rating is required, then we shouldn't need to apply a default value?

File: server/src/routes/api/v1/websitesCommentsRouter.js [hunk 4]
```diff
@@ -5,17 +5,18 @@ const websitesCommentsRouter = new express.Router({ mergeParams: true });
...
   try {
     const userId = req.user.id;
-    const newComment = await Comment.query().insert({ comment, userId, websiteId });
+    const newComment = await Comment.query().insert({ rating, comment, userId, websiteId });
 
     return res.status(201).json({ comment: newComment });
   } catch (error) {
     console.error(`ERROR ${error.message}`);
-    return res.status(500).json({ error: "Unable to add comment" });
+    return res.status(500).json({ error: "Unable to add comment" })
```
Comment (dovakeen118) [inline server/src/routes/api/v1/websitesCommentsRouter.js:17]: a POST endpoint requires proper error handling for ValidationErrors

**Review by dovakeen118 — COMMENTED** [review]
some small comments, largely proper validation error handling is needed for the POST request. That wasn't the main focus of this feature and fine if you don't tackle with the other updates, but something that should be top priority considering the rest of the POST functionality is there

**Comment by RipNasty** [comment]
We also updated the CommentSerializer so that all displayed comments would include our rating.

**Comment by Chepenik** [comment]
excellent work on this second go around David!

File: client/src/components/SocialMediaShow.js [hunk 5]
```diff
@@ -38,36 +34,43 @@ const SocialMediaShow = (props) => {
...
-  const commentList = comments && comments.length > 0 ? (
-    comments.map((comment, index) => (
-      <CommentTile key={comment.id} comment={comment} index={index} rating={comment.rating}/>
-    ))
-  ) : (
-    <p>No comments yet.</p>
-  );
+  const commentList =
+    socialMediaShow.comments && socialMediaShow.comments.length > 0 ? (
+      socialMediaShow.comments.map((comment, index) => (
```
Comment (dovakeen118) [inline client/src/components/SocialMediaShow.js:63]: can just check  length of comments to conditionally display message

File: server/src/routes/api/v1/websitesCommentsRouter.js [hunk 6]
```diff
@@ -10,10 +11,12 @@ websitesCommentsRouter.post("/", async (req, res) => {
   try {
     const userId = req.user.id;
     const newComment = await Comment.query().insert({ rating, comment, userId, websiteId });
-
     return res.status(201).json({ comment: newComment });
   } catch (error) {
     console.error(`ERROR ${error.message}`);
```
Comment (dovakeen118) [inline server/src/routes/api/v1/websitesCommentsRouter.js:16]: remove console logs from final code - this is fine to add while developing on the feature

File: client/src/components/SocialMediaShow.js [hunk 7]
```diff
@@ -1,68 +1,76 @@
...
+        <CommentTile
+          key={comment.id}
+          comment={comment}
+          index={index}
```
Comment (dovakeen118) [inline client/src/components/SocialMediaShow.js:67]: index is not used in the component - remove here and from map

**Review by dovakeen118 — COMMENTED** [review]
small changes!

**Pull request merged**